### ISSUE TITLE
[CI] AsyncEsqlQueryActionIT classMethod failing

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#106426]
**Build scan:**
https://gradle-enterprise.elastic.co/s/vtm2mlzovzqwe/tests/:x-pack:plugin:esql:internalClusterTest/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT

https://gradle-enterprise.elastic.co/s/vtm2mlzovzqwe/tests/:x-pack:plugin:esql:internalClusterTest/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT
https://gradle-enterprise.elastic.co/s/vtm2mlzovzqwe/tests/:x-pack:plugin:esql:internalClusterTest/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT/testFinishingBeforeTimeoutKeep
https://gradle-enterprise.elastic.co/s/vtm2mlzovzqwe/tests/task/:x-pack:plugin:esql:internalClusterTest/details/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT/testBasicAsyncExecution
https://gradle-enterprise.elastic.co/s/vtm2mlzovzqwe/tests/task/:x-pack:plugin:esql:internalClusterTest/details/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT?top-execution=1

**Reproduction line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT#classMethod`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('classMethod'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure excerpt:**
```
java.lang.IllegalStateException: Some shards are still open after the threadpool terminated. Something is leaking index readers or store references.

  at __randomizedtesting.SeedInfo.seed([D273A558C01BFDA9]:0)
  at org.elasticsearch.node.Node.awaitClose(Node.java:703)
  at org.elasticsearch.test.InternalTestCluster$NodeAndClient.close(InternalTestCluster.java:1061)
  at org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
  at org.elasticsearch.core.IOUtils.close(IOUtils.java:119)
  at org.elasticsearch.test.InternalTestCluster.close(InternalTestCluster.java:910)
  at org.elasticsearch.test.CloseableTestClusterWrapper.close(CloseableTestClusterWrapper.java:26)
  at org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
  at org.elasticsearch.core.IOUtils.close(IOUtils.java:119)
  at org.elasticsearch.test.ESIntegTestCase.clearClusters(ESIntegTestCase.java:547)
  at org.elasticsearch.test.ESIntegTestCase.afterClass(ESIntegTestCase.java:2336)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:909)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

**Comment by alex-spies** [comment]
Hi @ChrisHegarty , do you think you could assess whether that's a blocker?

I found this in the console log:
```
    org.elasticsearch.ResourceNotFoundException: sink exchanger for id [MowR3iSsTYuWFEkVoFOWQg:83] doesn't exist
        at org.elasticsearch.compute.operator.exchange.ExchangeService.getSinkHandler(ExchangeService.java:108)
        at org.elasticsearch.xpack.esql.plugin.ComputeService.runComputeOnDataNode(ComputeService.java:692)
        at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:753)
        at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:728)
        at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75)
        at org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:288)
        at org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:301)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(Thread.java:833)

    java.lang.AssertionError: Request breaker not reset to 0 on node: node_s2
    Expected: <0L>
         but: was <4132L>
```

**Comment by ChrisHegarty** [comment]
Seems like an intermittent failure, likely due to a race or something in test. Not a blocker.

**Comment by ChrisHegarty** [comment]
> Happened again - https://gradle-enterprise.elastic.co/s/hmtn5ojikvh3c/tests/:x-pack:plugin:esql:internalClusterTest/org.elasticsearch.xpack.esql.action.AsyncEsqlQueryActionIT/testFinishingBeforeTimeoutKeep

Exception from the above failure.

```
org.elasticsearch.ResourceNotFoundException: sink exchanger for id [WKqAsDezSrecr-tHpzudpg:85] doesn't exist |  

at __randomizedtesting.SeedInfo.seed([CBC2FD2D7D7C16EC:9BB72F18D4996A91]:0) |  
  | at org.elasticsearch.compute.operator.exchange.ExchangeService.getSinkHandler(ExchangeService.java:125) |  
  | at org.elasticsearch.xpack.esql.plugin.ComputeService.runComputeOnDataNode(ComputeService.java:714) |  
  | at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:787) |  
  | at org.elasticsearch.xpack.esql.plugin.ComputeService$DataNodeRequestHandler.messageReceived(ComputeService.java:754) |  
  | at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:75) |  
  | at org.elasticsearch.transport.TransportService$6.doRun(TransportService.java:1069) |  
  | at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984) |  
  | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) |  
  | at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33) |  
  | at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984) |  
  | at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) |  
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) |  
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) |  
  | at java.lang.Thread.run(Thread.java:1570)
```

**Comment by dnhatn** [comment]
> org.elasticsearch.ResourceNotFoundException: sink exchanger for id [WKqAsDezSrecr-tHpzudpg:85] doesn't exist |  

I think this is fixed in #110569, and I am closing this issue.
